pageserver/src/tenant/storage_layer/layer_desc.rs

```diff
@@ -103,6 +103,23 @@ impl PersistentLayerDesc {
         }
     }
 
+    pub fn from_filename(
+        tenant_id: TenantId,
+        timeline_id: TimelineId,
+        filename: LayerFileName,
+        file_size: u64,
+    ) -> Self {
+        match filename {
+            LayerFileName::Image(i) => {
+                // FIXME: can we incremental image layers or is this ffu?
+                Self::new_img(tenant_id, timeline_id, i.key_range, i.lsn, false, file_size)
+            }
+            LayerFileName::Delta(d) => {
+                Self::new_delta(tenant_id, timeline_id, d.key_range, d.lsn_range, file_size)
+            }
+        }
+    }
+
```

Comment:
This is extra.

Reply:
[413d669](https://github.com/neondatabase/neon/pull/5013/commits/413d669255bb21290103307c621b880b24ccbcef)